activerecord/lib/active_record/relation/batches.rb

```diff
@@ -215,72 +218,159 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
         act_on_ignored_order(error_on_ignore)
       end
 
+      keys = coerce_array(by)
+      raise ArgumentError, ":by must be present" if keys.empty?
+
+      start = coerce_array(start)
+      raise ArgumentError.new("Start length must not exceed by length") if start.length > keys.length
+
+      finish = coerce_array(finish)
+      raise ArgumentError.new("Finish length must not exceed by length") if finish.length > keys.length
```

Comment:
You might want to be consistent here with either `raise Error.new(msg)` or `raise Error, msg`.